rust/processor/src/worker.rs

```diff
@@ -739,6 +514,92 @@ impl Worker {
     }
 }
 
+async fn fetch_transactions(
+    processor_name: &str,
+    stream_address: &str,
+    receiver: kanal::AsyncReceiver<TransactionsPBResponse>,
+    task_index: usize,
+) -> TransactionsPBResponse {
+    let pb_channel_fetch_time = std::time::Instant::now();
+    let txn_pb_res = receiver.recv().await;
+    // Track how much time this thread spent waiting for a pb bundle
+    PB_CHANNEL_FETCH_WAIT_TIME_SECS
+        .with_label_values(&[processor_name, &task_index.to_string()])
+        .set(pb_channel_fetch_time.elapsed().as_secs_f64());
+
+    match txn_pb_res {
+        Ok(txn_pb) => txn_pb,
+        Err(_e) => {
+            error!(
+                processor_name = processor_name,
+                service_type = PROCESSOR_SERVICE_TYPE,
+                stream_address = stream_address,
+                "[Parser][T#{}] Consumer thread timed out waiting for transactions",
+                task_index
+            );
+            panic!(
+                "[Parser][T#{}] Consumer thread timed out waiting for transactions",
+                task_index
+            );
+        },
+    }
+}
+
+pub async fn do_processor(
+    transactions_pb: TransactionsPBResponse,
+    processor: &Processor,
+    db_chain_id: u64,
+    processor_name: &str,
+    auth_token: &str,
+    enable_verbose_logging: bool,
+) -> Result<ProcessingResult> {
+    let transactions_pb_slice = transactions_pb.transactions.as_slice();
+    let first_txn = transactions_pb_slice.first().unwrap();
+
+    let last_txn = transactions_pb_slice.last().unwrap();
+
+    let start_version = first_txn.version;
+    let end_version = last_txn.version;
+    let txn_time = first_txn.timestamp.clone();
+
+    if let Some(ref t) = txn_time {
+        PROCESSOR_DATA_RECEIVED_LATENCY_IN_SECS
+            .with_label_values(&[auth_token, processor_name])
+            .set(time_diff_since_pb_timestamp_in_secs(t));
+    }
```

Comment:
We can probably remove this metric since we already have `PB_CHANNEL_FETCH_WAIT_TIME_SECS`